src/Microsoft.Health.Fhir.Azure/IntegrationDataStore/AzureBlobSourceStream.cs

```diff
@@ -175,7 +176,12 @@ private async Task<Stream> DownloadDataFunc(long offset, long length)
         {
             // Stream is returned to the method caller, unable to dispose it under the current scope.
             var stream = new RecyclableMemoryStream(_recyclableMemoryStreamManager, tag: nameof(AzureBlobSourceStream));
-            await _blobClient.DownloadRangeToStreamAsync(stream, offset, length);
+            var options = new BlobDownloadOptions { Range = new HttpRange(offset, length) };
+
+            var response = await _blobClient.DownloadStreamingAsync(options);
```

Comment:
No methods in v12 API that takes an offset, a length, and stream to write to. DwonloadStreaing returns a stream and the content needs to be copied to another stream.